### ISSUE TITLE
Helm User Guide doesn't work well

### ISSUE THREAD

**Issue by pollosp** [operator-framework/operator-sdk#1614]
## Bug Report

**What did you do?**
Follow the example https://github.com/operator-framework/operator-sdk/blob/master/doc/helm/user-guide.md  step by step. 
But we notice that by default role didn't have permission to get pods. 

```
{"level":"error","ts":1561665041.4728038,"logger":"cmd","msg":"Failed to become leader.","Namespace":"default","error":"pods \"nginx-operator-6dff675cf9-vj98n\" is forbidden: User \"system:serviceaccount:default:nginx-operator\" cannot get resource \"pods\" in API group \"\" in the namespace \"default\"","stacktrace":"github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/helm.Run\n\tsrc/github.com/operator-framework/operator-sdk/pkg/helm/run.go:106\ngithub.com/operator-framework/operator-sdk/cmd/operator-sdk/run.newRunHelmCmd.func1\n\tsrc/github.com/operator-framework/operator-sdk/cmd/operator-sdk/run/helm.go:38\ngithub.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).execute\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:762\ngithub.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).ExecuteC\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:852\ngithub.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra.(*Command).Execute\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/spf13/cobra/command.go:800\nmain.main\n\tsrc/github.com/operator-framework/operator-sdk/cmd/operator-sdk/main.go:80\nruntime.main\n\t/home/travis/.gimme/versions/go1.12.5.linux.amd64/src/runtime/proc.go:200"}
```
We added full permissions in the role.yml
```
ccat deploy/role.yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  creationTimestamp: null
  name: nginx-operator
rules:
- apiGroups:
  - ""
  resources:
  - '*'
  verbs:
  - '*'
- apiGroups:
  - ""
  resources:
  - '*'
  verbs:
  - '*'
- apiGroups:
  - example.com
  resources:
  - '*'
  verbs:
  - '*'
```
And we go the following error and we get stuck, talking with people in the k8s slack recommend my open an issue here to handle the problem better.
```
{"level":"info","ts":1561665408.0445516,"logger":"cmd","msg":"Go Version: go1.12.5"}
{"level":"info","ts":1561665408.0446255,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1561665408.0446491,"logger":"cmd","msg":"Version of operator-sdk: v0.8.0"}
{"level":"info","ts":1561665408.0446806,"logger":"cmd","msg":"Watching single namespace.","Namespace":"default"}
{"level":"info","ts":1561665408.1029792,"logger":"kubebuilder.controller","msg":"Starting EventSource","controller":"nginx-controller","source":"kind source: example.com/v1alpha1, Kind=Nginx"}
{"level":"info","ts":1561665408.1035173,"logger":"helm.controller","msg":"Watching resource","apiVersion":"example.com/v1alpha1","kind":"Nginx","namespace":"default","reconcilePeriod":"1m0s"}
{"level":"info","ts":1561665408.103629,"logger":"leader","msg":"Trying to become the leader."}
{"level":"info","ts":1561665408.15045,"logger":"leader","msg":"Found existing lock","LockOwner":"nginx-operator-6dff675cf9-fw2hz"}
{"level":"info","ts":1561665408.158502,"logger":"leader","msg":"Not the leader. Waiting."}
{"level":"info","ts":1561665409.2859418,"logger":"leader","msg":"Became the leader."}
{"level":"info","ts":1561665409.3878603,"logger":"kubebuilder.controller","msg":"Starting Controller","controller":"nginx-controller"}
{"level":"info","ts":1561665409.4886916,"logger":"kubebuilder.controller","msg":"Starting workers","controller":"nginx-controller","worker count":1}
2019/06/27 19:56:49 uninstall: Failed deletion of "example-nginx-456olflgn9atwdw06jrhnqz9f": deployments.apps "example-nginx-456olflgn9atwdw06jrhnqz9f" is forbidden: User "system:serviceaccount:default:nginx-operator" cannot delete resource "deployments" in API group "apps" in the namespace "default"
{"level":"error","ts":1561665409.7502625,"logger":"helm.controller","msg":"Failed to install release","namespace":"default","name":"example-nginx","apiVersion":"example.com/v1alpha1","kind":"Nginx","release":"example-nginx-456olflgn9atwdw06jrhnqz9f","error":"release example-nginx-456olflgn9atwdw06jrhnqz9f failed: deployments.apps is forbidden: User \"system:serviceaccount:default:nginx-operator\" cannot create resource \"deployments\" in API group \"apps\" in the namespace \"default\"","stacktrace":"github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/operator-framework/operator-sdk/vendor/github.com/go-logr/zapr/zapr.go:128\ngithub.com/operator-framework/operator-sdk/pkg/helm/controller.HelmOperatorReconciler.Reconcile\n\tsrc/github.com/operator-framework/operator-sdk/pkg/helm/controller/reconcile.go:172\ngithub.com/operator-framework/operator-sdk/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\tsrc/github.com/operator-framework/operator-sdk/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:215\ngithub.com/operator-framework/operator-sdk/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1\n\tsrc/github.com/operator-framework/operator-sdk/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:158\ngithub.com/operator-framework/operator-sdk/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\tsrc/github.com/operator-framework/operator-sdk/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133\ngithub.com/operator-framework/operator-sdk/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil\n\tsrc/github.com/operator-framework/operator-sdk/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:134\ngithub.com/operator-framework/operator-sdk/vendor/k8s.io/apimachinery/pkg/util/wait.Until\n\tsrc/github.com/operator-framework/operator-sdk/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88"}
```
**What did you expect to see?**
A working Nginx operator

**What did you see instead? Under which circumstances?**
Deployed operator but it doesn't work because it doesn't create the nginx pod when we request them.

**Environment**
* operator-sdk version:

operator-sdk version: v0.8.0, commit: 78c472461e75e6c64589cfadf577a2004b8a26b3

* go version:

go version go1.12.5 darwin/amd64

* Kubernetes version information:
Kubernetes  v1.14.2 using Kind version v0.3.0

* Kubernetes cluster kind: 

* Are you writing your operator in ansible, helm, or go?
HELM

**Possible Solution**

**Additional context**

**Comment by joelanford** [comment]
I think your primary issue is that `0.8.0` has a known bug with the Helm role generation, which is fixed in [`0.8.1`](https://github.com/operator-framework/operator-sdk/releases/tag/v0.8.1) . Can you upgrade to `0.8.1` and try the Helm user guide again from the beginning? The bug was in `operator-sdk new` when it generates the default `role.yaml`.

**Comment by pollosp** [comment]
@joelanford  working like a charm :) 
```
✗ kubectl get pods
NAME                                                      READY   STATUS    RESTARTS   AGE
example-nginx-bvxil3q6d7pfymdilaxgj4jr7-f5f8847d5-72225   1/1     Running   0          64s
example-nginx-bvxil3q6d7pfymdilaxgj4jr7-f5f8847d5-vbmkb   1/1     Running   0          64s
nginx-operator-6dff675cf9-php9h                           1/1     Running   0          118s
```
